### ISSUE TITLE
Yielding bytes and streaming api

### ISSUE THREAD

**Issue by patarapolw** [pndurette/gTTS#318]
## Prerequisites
* [x] Did you make sure a similar [issue](../) didn't exist?
* [x] Did you update gTTS to the latest? (`pip install --upgrade gTTS`)

## Proposed Behaviour

It would nice if you make streaming API for gTTS that circumvent the 100 characters' limit (https://github.com/pndurette/gTTS/issues/1). That is, making this kind of code send to client earlier, even with long querystring

```python
import re
from pathlib import Path

from fastapi import FastAPI
from fastapi.responses import StreamingResponse

from gtts import gTTS

app = FastAPI()

@app.get("/api/tts")
def api_tts(q: str, lang: str):
    safe_q = re.sub(r"\W+", q, " ")
    cache_path = "static"
    save_path = Path(cache_path, lang, safe_q + ".mp3")

    if not save_path.exists():
        Path(cache_path, lang).mkdir(exist_ok=True)
        gTTS(q, lang=lang).save(str(save_path))

    def iter_file():
        with save_path.open("rb") as f:
            yield from f

    return StreamingResponse(iter_file(), media_type="audio/mp3")
```

## Context

In an old library in Node.js NPM, this kind of code is possible.

```js
var express = require('express');
var router = express.Router();
var gtts = require('node-gtts')('en');
 
router.get('/speech', function(req, res) {
  res.set({'Content-Type': 'audio/mpeg'});
  gtts.stream(req.query.text).pipe(res);
})
```

**Comment by pndurette** [comment]
Oh interesting! Thank you for this, I'll take a look this week!

**Comment by pndurette** [comment]
Thanks for contribution and your very long patience! Released in `gTTS 2.2.4`!
